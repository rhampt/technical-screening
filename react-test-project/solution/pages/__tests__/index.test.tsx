import renderer from 'react-test-renderer'
import { shallow, mount, render } from 'enzyme'
import Home from '../index'

describe('Home page', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<Home />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
