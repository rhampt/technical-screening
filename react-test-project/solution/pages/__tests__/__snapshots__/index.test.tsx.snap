// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should match the snapshot 1`] = `
<div
  className="container"
>
  <main
    className="main"
  >
    <h1
      className="title"
    >
      Weather
    </h1>
    <div
      className="content"
    >
      <div
        className="formContainer"
      >
        <div
          className="MuiFormControl-root state formField"
        >
          <div
            aria-expanded={false}
            aria-owns={null}
            className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="combobox"
            style={
              Object {
                "width": 300,
              }
            }
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              onKeyPress={[Function]}
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="combo-box-demo"
                id="combo-box-demo-label"
              >
                State
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                onClick={[Function]}
              >
                <input
                  aria-activedescendant={null}
                  aria-autocomplete="list"
                  aria-controls={null}
                  aria-invalid={false}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                  disabled={false}
                  id="combo-box-demo"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  required={false}
                  spellCheck="false"
                  type="text"
                  value=""
                />
                <div
                  className="MuiAutocomplete-endAdornment"
                >
                  <button
                    aria-label="Clear"
                    className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    title="Clear"
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="Open"
                    className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    title="Open"
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-3"
                  >
                    <span>
                      State
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiFormControl-root formField"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
            onKeyPress={[Function]}
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="outlined-basic"
              id="outlined-basic-label"
            >
              City
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="outlined-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    City
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiFormControl-root formField"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
            onKeyPress={[Function]}
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="outlined-basic"
              id="outlined-basic-label"
            >
              ZIP
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="outlined-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    ZIP
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiFormControl-root tempUnit formField"
        >
          <label
            className="MuiFormLabel-root"
          >
            Unit:
          </label>
          <div
            aria-label="temperature unit"
            className="MuiToggleButtonGroup-root"
            role="group"
          >
            <button
              aria-label="celsius"
              aria-pressed={false}
              className="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-sizeSmall"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
              value="metric"
            >
              <span
                className="MuiToggleButton-label"
              >
                <span>
                  °C
                </span>
              </span>
            </button>
            <button
              aria-label="fahrenheit"
              aria-pressed={true}
              className="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal Mui-selected MuiToggleButton-sizeSmall"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
              value="imperial"
            >
              <span
                className="MuiToggleButton-label"
              >
                <span>
                  °F
                </span>
              </span>
            </button>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained submit MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Show
          </span>
        </button>
      </div>
      
    </div>
  </main>
  <footer
    className="footer"
  >
    Powered by Voomove
  </footer>
</div>
`;
